Autopilot/AttitudeManager/Drivers/RadioXbee.c

```diff
@@ -268,6 +276,9 @@ uint8_t* parseUplinkPacket(uint16_t* length)
 
     //the parsed length of the packet we're parsing. Used to tell when the packet is complete
     static uint16_t rx_packet_length;
+    
+    //current calculated checksum of the in-progress packet
+    static uint8_t checksum;
```

Comment:
Because when this method gets called we might not have received the entire xbee frame. This method must be capable of handling of parsing partial xbee frames, and so we need to control when the checksum gets reset, and also keep its state throughout the receival of the packet